test/batch-submitter/batch-submitter.spec.ts

```diff
@@ -0,0 +1,544 @@
+        expect(receipt).to.be.undefined
+      })
+
+      it('should submit if gas price is not over threshold', async () => {
+        l2Provider.setNumBlocksToReturn(2)
+        l2Provider.setL2BlockData({
+          queueOrigin: QueueOrigin.L1ToL2,
+        } as any)
+
+        const lowGasPriceWei = BigNumber.from(2).mul(1_000_000_000)
+
+        sinon
+          .stub(sequencer, 'getGasPrice')
+          .callsFake(async () => lowGasPriceWei)
+
+        const receipt = await batchSubmitter.submitNextBatch()
+        expect(sequencer.getGasPrice).to.have.been.calledOnce
+        expect(receipt).to.not.be.undefined
+      })
+    })
+  })
+
+  describe('StateBatchSubmitter', () => {
+    let txBatchSubmitter
+    let stateBatchSubmitter
+    beforeEach(async () => {
+      for (let i = 1; i < 15; i++) {
+        await OVM_CanonicalTransactionChain.enqueue(
+          '0x' + '01'.repeat(20),
+          50_000,
+          '0x' + i.toString().repeat(64),
+          {
+            gasLimit: 1_000_000,
+          }
+        )
+      }
+
+      txBatchSubmitter = createBatchSubmitter(0)
+
+      l2Provider.setNumBlocksToReturn(5)
+      const nextQueueElement = await getQueueElement(
+        OVM_CanonicalTransactionChain
+      )
+      const data = ctcCoder.eip155TxData.encode({
+        sig: DUMMY_SIG,
+        gasLimit: 0,
+        gasPrice: 0,
+        nonce: 0,
+        target: DUMMY_ADDRESS,
+        data: '0x',
+        type: TxType.EIP155,
+      })
+      l2Provider.setL2BlockData(
+        {
+          data,
+          l1BlockNumber: nextQueueElement.blockNumber - 1,
+          txType: TxType.EIP155,
+          queueOrigin: QueueOrigin.Sequencer,
+          l1TxOrigin: null,
+        } as any,
+        nextQueueElement.timestamp - 1,
+        EXAMPLE_STATE_ROOT // example stateRoot
+      )
+
+      // submit a batch of transactions to enable state batch submission
+      const txReceipt = await txBatchSubmitter.submitNextBatch()
+
+      stateBatchSubmitter = new StateBatchSubmitter(
+        sequencer,
+        l2Provider as any,
+        MIN_TX_SIZE,
+        MAX_TX_SIZE,
+        10, // maxBatchSize
+        0,
+        1,
+        100000,
+        0, // finalityConfirmations
+        AddressManager.address,
+        1,
+        MIN_GAS_PRICE_IN_GWEI,
+        MAX_GAS_PRICE_IN_GWEI,
+        GAS_RETRY_INCREMENT,
+        GAS_THRESHOLD_IN_GWEI,
+        new Logger({ name: STATE_BATCH_SUBMITTER_LOG_TAG }),
+        '0x' + '01'.repeat(20) // placeholder for fraudSubmissionAddress
+      )
+    })
+
+    describe('submitNextBatch', () => {
+      it('should submit a state batch after a transaction batch', async () => {
+        const receipt = await stateBatchSubmitter.submitNextBatch()
+        const iface = new ethers.utils.Interface(scc.abi)
+        const parsedLogs = iface.parseLog(receipt.logs[0])
+
+        expect(parsedLogs.eventFragment.name).to.eq('StateBatchAppended')
```

Comment:
would love feedback on what else to test here!

Reply:
modeled the following tests after what we test for in transaction batch submission